### PR TITLE
[CAPI]Fix some potential segment fault in test cases

### PR DESCRIPTION
**Pull request by riverlijunjie**
### Details:
 - Fix some potential segment fault or crash in test cases

### Tickets:
 - 92079

### CODE REVIEW

File: src/bindings/c/tests/ov_infer_request_test.cpp [hunk 1]
```diff
@@ -8,20 +8,20 @@
 inline void get_tensor_info(ov_model_t* model, bool input, char** name, ov_shape_t* shape, ov_element_type_e* type) {
     ov_output_const_port* port = nullptr;
     if (input) {
-        OV_EXPECT_OK(ov_model_const_input(model, &port));
+        OV_ASSERT_OK(ov_model_const_input(model, &port));
```
Comment (ilyachur) [inline src/bindings/c/tests/ov_infer_request_test.cpp:11]: @riverlijunjie I think we shouldn't always use `ASSERT`.
`ASSERT` is needed if we want to finish test execution after the fail (for example after that we cannot work with incorrect object), but if we want just to check the value which doesn't impact future execution, it is better to use `EXPECT` in order to get all errors and don't stop on the first one.

Comment (riverlijunjie) [reply src/bindings/c/tests/ov_infer_request_test.cpp:11]: ok, let's check which cases need use `EXCEPT `and which need use `ASSERT`.

File: src/bindings/c/tests/ov_compiled_model_test.cpp [hunk 2]
```diff
@@ -12,15 +12,15 @@ TEST_P(ov_compiled_model, ov_compiled_model_inputs_size) {
     ASSERT_NE(nullptr, core);
 
     ov_model_t* model = nullptr;
-    OV_EXPECT_OK(ov_core_read_model(core, xml, bin, &model));
+    OV_ASSERT_OK(ov_core_read_model(core, xml, bin, &model));
```
Comment (ilya-lavrenov) [inline src/bindings/c/tests/ov_compiled_model_test.cpp:15]: why do we need assert here? I suppose this is ok if we don't have fatal failure here.

Comment (riverlijunjie) [reply src/bindings/c/tests/ov_compiled_model_test.cpp:15]: If read_model failed, the following calling(depends on ov_model) will be invalid.

Comment (ilyachur) [reply src/bindings/c/tests/ov_compiled_model_test.cpp:15]: @riverlijunjie In case of fails in read_model, can we always set `ov_model` to NULL?

Comment (riverlijunjie) [reply src/bindings/c/tests/ov_compiled_model_test.cpp:15]: yes, we have initialized ov_model to be null before call ov_core_read_model(), so if it failed, it is still null.

Comment (ilya-lavrenov) [reply src/bindings/c/tests/ov_compiled_model_test.cpp:15]: in this case we don't see assert, we should use expect. 
Please, read about different for gtest assert and expect.

Comment (riverlijunjie) [reply src/bindings/c/tests/ov_compiled_model_test.cpp:15]: ok, I follow the gtest convention.

Comment (riverlijunjie) [reply src/bindings/c/tests/ov_compiled_model_test.cpp:15]: Some items about "EXPECT" and "ASSERT" in gtest:

Use ASSERT when the condition must hold - if it doesn't the test stops right there. Use this when the remainder of the test doesn't have semantic meaning without this condition holding.

Use EXPECT when the condition should hold, but in cases where it doesn't we can still get value out of continuing the test. (The test will still ultimately fail at the end, though.)

The rule of thumb is: use EXPECT by default, unless you require something to hold for the remainder of the tests, in which case you should use ASSERT for that particular condition.

**Comment by ilya-lavrenov** [comment]
I run the tests and still observe a crash:
```
[ RUN      ] device_name/ov_core.ov_core_set_property_enum/0
[       OK ] device_name/ov_core.ov_core_set_property_enum/0 (0 ms)
[ RUN      ] device_name/ov_core.ov_core_set_property_invalid_number_property_arguments/0
/Users/sandye51/Documents/Programming/git_repo/openvino/src/bindings/c/tests/ov_core_test.cpp:200: Failure
Expected equality of these values:
  ov_status_e::OK
    Which is: 0
  ov_core_get_property(core, device_name.c_str(), key_2, &ret)
    Which is: -17
ov_capi_test(96574,0x1032c4580) malloc: *** error for object 0x6000011f11a0: pointer being freed was not allocated
ov_capi_test(96574,0x1032c4580) malloc: *** set a breakpoint in malloc_error_break to debug
zsh: abort      ./bin/arm64/Release/ov_capi_test
```

**Comment by riverlijunjie** [comment]
> I run the tests and still observe a crash:
> 
> ```
> [ RUN      ] device_name/ov_core.ov_core_set_property_enum/0
> [       OK ] device_name/ov_core.ov_core_set_property_enum/0 (0 ms)
> [ RUN      ] device_name/ov_core.ov_core_set_property_invalid_number_property_arguments/0
> /Users/sandye51/Documents/Programming/git_repo/openvino/src/bindings/c/tests/ov_core_test.cpp:200: Failure
> Expected equality of these values:
>   ov_status_e::OK
>     Which is: 0
>   ov_core_get_property(core, device_name.c_str(), key_2, &ret)
>     Which is: -17
> ov_capi_test(96574,0x1032c4580) malloc: *** error for object 0x6000011f11a0: pointer being freed was not allocated
> ov_capi_test(96574,0x1032c4580) malloc: *** set a breakpoint in malloc_error_break to debug
> zsh: abort      ./bin/arm64/Release/ov_capi_test
> ```

If ov::hint::performance property is not set before, we call get_property to get its value, it should return `ov::hint::PerformanceMode::UNDEFINED`.  It will be converted to string format, the string value is 0 (string_value=0, not "UNDEFINED"):

```
    ov::Any test = ov::hint::PerformanceMode::UNDEFINED;
    std::cout << "test = " << test.as<std::string>() << std::endl;
```

It will print out: `test = `

In x86, it didn't report any problem, but in arm as @ilya-lavrenov shared, it will encounter problem during copying it to char array.
To avoid this behavior in arm during convert ov::hint::PerformanceMode::UNDEFINED to string format, we can modify the test case.

File: src/bindings/c/tests/ov_core_test.cpp [hunk 3]
```diff
@@ -171,33 +171,33 @@ TEST_P(ov_core, ov_core_compile_model_from_file) {
...
-    OV_ASSERT_OK(ov_core_create(&core));
+    OV_EXPECT_OK(ov_core_create(&core));
     ASSERT_NE(nullptr, core);
 
     const char* key_1 = ov_property_key_inference_num_threads;
     const char* value_1 = "12";
-    const char* key_2 = ov_property_key_hint_performance_mode;
-    const char* value_2 = "LATENCY";
+    const char* key_2 = ov_property_key_num_streams;
+    const char* value_2 = "7";
```
Comment (ilya-lavrenov) [inline src/bindings/c/tests/ov_core_test.cpp:193]: now ARM supports performance mode, does it allow to revert this line to LATENCY?

Comment (riverlijunjie) [reply src/bindings/c/tests/ov_core_test.cpp:193]: No problem! Does CI enable arm platform test cases so that we can have a test?

**Pull request merged**